### PR TITLE
Add example project

### PR DESCRIPTION
**Pull request by kpfromer**
Add a proper example nestjs project using nestjs-typegoose for documentation and testing purposes.

### CODE REVIEW

File: example/src/cat/cats.service.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { ReturnModelType } from '@typegoose/typegoose';
+
+@Injectable()
+export class CatsService {
+  constructor(
+    @InjectModel(Cat) private readonly catModel: ReturnModelType<typeof Cat>,
+  ) {}
+
+  async create(createCatDto: { name: string }): Promise<Cat> {
+    const createdCat = new this.catModel(createCatDto);
```
Comment (hasezoey) [inline example/src/cat/cats.service.ts]:
```suggestion
    return await this.catModel.create(createCatDto);
```

File: example/src/cat/cats.service.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    return await createdCat.save();
```
Comment (hasezoey) [inline example/src/cat/cats.service.ts]:
```suggestion
```

File: example/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+
+async function bootstrap() {
```
Comment (hasezoey) [inline example/src/main.ts]: should this be changed to an [IIFE](https://wiki.selfhtml.org/wiki/IIFE)?

File: example/src/cat/cats.service.ts (hunk 1 above)
Comment (kpfromer) [reply example/src/cat/cats.service.ts]: The docs should probably be updated as well.

**Pull request merged**